src/FOM/UserBundle/Component/FOMIdentitiesProvider.php

```diff
@@ -14,103 +16,124 @@
         $groups = $repo->findAll();
 
         $roles = array();
-        foreach($groups as $group) {
+        foreach ($groups as $group) {
             $roles[] = 'r:' . $group->getAsRole();
         }
 
         return $roles;
     }
 
-    public function getAllGroups(){
-        $repo = $this->getDoctrine()->getRepository('FOMUserBundle:Group');
+    public function getAllGroups()
+    {
+        $all = array();
+        if ($this->container->hasParameter('ldap_host')) {
+            $groupDn        = $this->container->getParameter('ldap_group_dn');
+            $ldapClient   = $this->getLdapClient();
+            $ldapGroupList = $ldapClient->find($groupDn, '(objectClass=top)');
+            if ($ldapGroupList != null) {
+                foreach (array_slice($ldapGroupList, 2) as $ldapGroup) {
+                    $group =    new Group();
+                    $group->setTitle( $ldapGroup['cn'][0]);
+                    $all[] = $group;
+                }
+
+            }
+
+        }
+
+
+        $repo   = $this->getDoctrine()->getRepository('FOMUserBundle:Group');
         $groups = $repo->findAll();
 
-        $all = array();
-        foreach($groups as $group) {
+        foreach ($groups as $group) {
+
             $all[] = $group;
         }
-
         return $all;
     }
 
-    public function getAllUsers(){
+    public function getAllUsers()
+    {
         // Settings for LDAP
-        if($this->container->hasParameter('ldap_host')) {
-            $ldapHostname = $this->container->getParameter("ldap_host");
-            $ldapPort = $this->container->getParameter("ldap_port");
-            $ldapVersion = $this->container->getParameter("ldap_version");
-            $baseDn = $this->container->getParameter("ldap_user_base_dn");
-            $nameAttribute = $this->container->getParameter("ldap_user_name_attribute");
-            $bindDn = $this->container->getParameter("ldap_bind_dn");
-            $bindPasswd = $this->container->getParameter("ldap_bind_pwd");
-            $filter = "(" . $nameAttribute . "=*)";
-
-            $connection = @ldap_connect($ldapHostname, $ldapPort);
-            ldap_set_option($connection, LDAP_OPT_PROTOCOL_VERSION, $ldapVersion);
-
-            if (strlen($bindDn) !== 0 && strlen($bindPasswd) !== 0) {
-                if (!ldap_bind($connection, $bindDn, $bindPasswd)) {
-                    throw new \Exception('Unable to bind LDAP to DN: ' . $bindDn);
+        $all = array();
+        if ($this->container->hasParameter('ldap_host')) {
+            $nameAttribute = $this->container->getParameter('ldap_group_name_attribute');
+            $userDn        = $this->container->getParameter('ldap_user_base_dn');
+
+            $ldapClient   = $this->getLdapClient();
+            $ldapUserList = $ldapClient->find($userDn, '(objectclass=top)');
+            if ($ldapUserList !=  null) {
+                unset($ldapUserList[0]);
+                foreach (array_slice($ldapUserList, 1) as $ldapUser) {
+
```

Comment:
You remove the first, then you slice the second entry of the results. Is this intended? Is this correct? Even if it is, it could simply be ```array_slice($ldapUserList, 2)```.
LdapClientInterface->find (upstream) has poor documentation, maybe explain here why you have to slice elements at all.